module/VuFind/src/VuFind/RecordDriver/MarcAdvancedTrait.php

```diff
@@ -635,29 +618,29 @@ public function getAllRecordLinks()
      * into the field number where relevant to generate a note to associate
      * with a record link.
      *
-     * @param File_MARC_Data_Field $field Field to examine
+     * @param array $field Field to examine
      *
      * @return string
      */
     protected function getRecordLinkNote($field)
     {
         // If set, use relationship information from subfield i
-        if ($subfieldI = $field->getSubfield('i')) {
+        if ($subfieldI = $this->getSubfield($field, 'i')) {
             // VuFind will add a colon to the label, so prevent double colons:
-            $data = rtrim(trim($subfieldI->getData(), ':'));
+            $data = rtrim($subfieldI, ':');
```

Comment:
Is there a reason you removed the internal whitespace trim? (I'm honestly not sure why that's needed in the first place... but just noticed it as a possible change in logic).

Reply:
getSubfield method already trims the value, so there's no need to do it again.